src/project_model/_unit_tests_discovery.rs

```diff
@@ -0,0 +1,139 @@
+//! Code in this file concerns the discovery of unit tests in a project.
+//!
+//! Current implementation is based on the mapping Cargo metadata targets to BSP build targets.
+//! Meaning that unit tests are not considered as a separate build targets.
+//! Therefore unit test execution is based on running tests on BSP build targets,
+//! which runs all the unit tests within BSP build target,
+//! including those written in files on which build target depends.
+//! <br><br>
+//! Unit tests discovery is possible using
+//!
+//! *`cargo +nightly test -- --list --format json -Zunstable-options`*
+//!
+//! however, this requires compilation of whole project, which is not desirable -
+//! it is time consuming and generates additional files.
+//!
+//! Command mentioned above discovers all tests, also *integration tests*,
+//! which are treated as a build targets by 'cargo metadata', so they need to be skipped or merged
+//! with the ones discovered in [ProjectWorkspace::new()](crate::project_model::workspace::ProjectWorkspace::new).
+//!
+//! Moreover, this approach gives us the information whether tests are ignored or not.
+//! Not achieved with 'cargo metadata'.
+//!
+//! **Alternative next steps:** Analyze how cargo discovers tests and compile only
+//! to the moment when unit tests can be obtained.
+
+use crate::bsp_types::RustBuildTargetData::Rust;
+use crate::bsp_types::{
+    BuildTarget, BuildTargetCapabilities, BuildTargetIdentifier, RustBuildTarget, RUST_ID,
+};
+use cargo_metadata::camino::Utf8PathBuf;
+use cargo_metadata::Edition;
+use serde::{Deserialize, Serialize};
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+
+/// Struct for deserialization of test JSON from
+/// 'cargo +nightly test -- --list --format json -Zunstable-options' output
+///
+/// Deserialization truncates discovered test JSON as below
```

Comment:
a czemu w sumie wywalacie te dane? one się na pewno przydadzą w którymś momencie później

Reply:
Mogą się przydać ale to jest i tak w całości plik, z którego wcale nie korzystamy i tylko baza pod wykrywanie unit testów, które zdecydowaliśmy się pominąć. 

Jeżeli ktoś zdecyduje się w przyszłości wykrywać jednak unit-testy, to będzie analizował dalej, czy reszta danych jest potrzebna, dlatego też wspomniałem o tym w komentarzu